File: data/readable_format/fortran/file367.f90
PROGRAM ezcu_test_fortran
    IMPLICIT NONE
    CALL handler_start()
    CALL base_test()
    CALL dev_test()
    CALL mem_test()
    CALL knl_test()
    CALL handler_end()
    CALL handler_stat()
END PROGRAM ezcu_test_fortran

SUBROUTINE handler_start()
    PRINT *, "Handler start."
END SUBROUTINE handler_start

SUBROUTINE base_test()
    PRINT *, "Base test."
END SUBROUTINE base_test

SUBROUTINE dev_test()
    PRINT *, "Device test."
END SUBROUTINE dev_test

SUBROUTINE mem_test()
    PRINT *, "Memory test."
END SUBROUTINE mem_test

SUBROUTINE knl_test()
    PRINT *, "Kernel test."
END SUBROUTINE knl_test

SUBROUTINE handler_end()
    PRINT *, "Handler end."
END SUBROUTINE handler_end

SUBROUTINE handler_stat()
    PRINT *, "Handler statistics."
END SUBROUTINE handler_stat

File: data/readable_format/cpp/file367.cpp
#include <iostream>

// Assuming dummy implementations of the previously separate module functionalities
namespace m_base_test {
    void base_test() {
        std::cout << "Base test." << std::endl;
    }
}

namespace m_dev_test {
    void dev_test() {
        std::cout << "Device test." << std::endl;
    }
}

namespace m_mem_test {
    void mem_test() {
        std::cout << "Memory test." << std::endl;
    }
}

namespace m_knl_test {
    void knl_test() {
        std::cout << "Kernel test." << std::endl;
    }
}

class m_handler {
public:
    static void handler_start() {
        std::cout << "Handler start." << std::endl;
    }

    static void handler_end() {
        std::cout << "Handler end." << std::endl;
    }

    static void handler_stat() {
        std::cout << "Handler statistics." << std::endl;
    }
};

int main() {
    // Using the functionalities directly in the main function
    m_handler::handler_start();

    m_base_test::base_test();
    m_dev_test::dev_test();
    m_mem_test::mem_test();
    m_knl_test::knl_test();

    m_handler::handler_end();
    m_handler::handler_stat();

    return 0;
}